File: TextIndexApi/src/main/java/com/hideakin/textsearch/index/model/UserInfo.java
package com.hideakin.textsearch.index.model;

import java.time.ZonedDateTime;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.hideakin.textsearch.index.entity.UserEntity;

public class UserInfo {

	@JsonProperty("uid")
	private int uid;
	
	@JsonProperty("username")
	private String username;
	
	@JsonProperty("roles")
	private String[] roles;
	
	@JsonProperty("created_at")
	private ZonedDateTime createdAt;
	
	@JsonProperty("updated_at")
	private ZonedDateTime updatedAt;
	
	@JsonProperty("access_token")
	private String accessToken;
	
	@JsonProperty("expires_at")
	private ZonedDateTime expiresAt;
	
	public UserInfo(UserEntity entity) {
		this.uid = entity.getUid();
		this.username = entity.getUsername();
		this.setRoles(entity.getRoles());
		this.createdAt = entity.getCreatedAt();
		this.updatedAt = entity.getUpdatedAt();
		this.accessToken = entity.getAccessToken();
		this.expiresAt = entity.getExpiresAt();
	}

	public int getUid() {
		return uid;
	}

	public void setUid(int uid) {
		this.uid = uid;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String[] getRoles() {
		return roles;
	}

	public void setRoles(String[] roles) {
		this.roles = roles;
	}

	public void setRoles(String roles) {
		this.roles = roles != null ? roles.split(",") : new String[0];
	}
	
	public boolean isAdministrator() {
		for (String role : roles) {
			if (role.equalsIgnoreCase("administrator")) {
				return true;
			}
		}
		return false;
	}

	public ZonedDateTime getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(ZonedDateTime createdAt) {
		this.createdAt = createdAt;
	}

	public ZonedDateTime getUpdatedAt() {
		return updatedAt;
	}

	public void setUpdatedAt(ZonedDateTime updatedAt) {
		this.updatedAt = updatedAt;
	}

	public String getAccessToken() {
		return accessToken;
	}

	public void setAccessToken(String accessToken) {
		this.accessToken = accessToken;
	}

	public ZonedDateTime getExpiresAt() {
		return expiresAt;
	}

	public void setExpiresAt(ZonedDateTime expiresAt) {
		this.expiresAt = expiresAt;
	}

}

File: TextSearchCLI/model/UserInfo.cs
using System;
using System.Text;
using Newtonsoft.Json;
using com.hideakin.textsearch.utility;

namespace com.hideakin.textsearch.model
{
    internal class UserInfo
    {
		private static readonly string DTFMT = "yyyy-MM-ddTHH:mm:ss.fff";

		[JsonProperty("uid")]
		public int Uid { get; set; }

		[JsonProperty("username")]
		public string Username { get; set; }

		[JsonProperty("roles")]
		public string[] Roles { get; set; }

		[JsonProperty("created_at")]
		public DateTime CreatedAt { get; set; }

		[JsonProperty("updated_at")]
		public DateTime UpdatedAt { get; set; }

		[JsonProperty("access_token")]
		public string AccessToken { get; set; }

		[JsonProperty("expires_at")]
		public DateTime? ExpiresAt { get; set; }

		public string RolesString
		{
			get
			{
				var sb = new StringBuilder();
				if (Roles.Length > 0)
				{
					sb.Append(Roles[0]);
					for (int index = 1; index < Roles.Length; index++)
					{
						sb.Append(",");
						sb.Append(Roles[index]);
					}
				}
				return sb.ToString();
			}
		}

        public override string ToString()
        {
			if (ExpiresAt != null)
			{
				return string.Format("uid={0} username={1} roles={2} created={3} updated={4} token={5} expires={6}",
					Uid, Username, RolesString, CreatedAt.ToString(DTFMT), UpdatedAt.ToString(DTFMT),
					AccessToken, ExpiresAt.Value.ToString(DTFMT));
			}
			else
			{
				return string.Format("uid={0} username={1} roles={2} created={3} updated={4} token= expires=",
					Uid, Username, RolesString, CreatedAt.ToString(DTFMT), UpdatedAt.ToString(DTFMT));
			}
        }
    }
}
